geo-types/src/lib.rs

```diff
@@ -81,6 +81,133 @@ mod macros;
 #[doc(hidden)]
 pub mod private_utils;
 
+impl<T: num_traits::Float + arbitrary::Arbitrary + CoordinateType> arbitrary::Arbitrary
+    for Coordinate<T>
+{
+    fn arbitrary(u: &mut arbitrary::Unstructured<'_>) -> arbitrary::Result<Self> {
+        let x = u.arbitrary::<T>()?;
+        if x.is_nan() {
+            return Err(arbitrary::Error::IncorrectFormat);
+        }
+
+        let y = u.arbitrary::<T>()?;
+        if y.is_nan() {
```

Comment:
I wonder if there would actually be some value in fuzzing with NaN some times. 

Generally we expect valid input, but ultimately our methods could be downstream from something that outputs NaN, and we should try to respond as best we can (e.g. probably not crash).